### PR TITLE
CloudRetry/AWSRetry : Disable catching of NotFound exceptions

### PR DESCRIPTION
**Pull request by tremble**
##### SUMMARY

AWSRetry auto-catches a wide variety of NotFound exceptions, which we don't always want to catch.

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

lib/ansible/module_utils/cloud.py
lib/ansible/module_utils/ec2.py

##### ADDITIONAL INFORMATION

see also #67204

There will be a number of followup PRs explicitly adding the various NotFound error codes to modules where we believe this behavior was expected.

### CODE REVIEW

File: changelogs/fragments/67281-AWSRetry-disable-NotFound.yaml
Comment (felixfontein) [inline changelogs/fragments/67281-AWSRetry-disable-NotFound.yaml]:
```suggestion
- "The ``AWSRetry`` decorator no longer catches ``NotFound`` exceptions unless they're explicitly added via ``catch_extra_error_codes``."
```

File: docs/docsite/rst/porting_guides/porting_guide_2.10.rst
Comment (felixfontein) [inline docs/docsite/rst/porting_guides/porting_guide_2.10.rst]:
```suggestion
* The ``AWSRetry`` decorator no longer catches ``NotFound`` exceptions by default. ``NotFound`` exceptions need to be explicitly added using ``catch_extra_error_codes``.  Some AWS modules may see an increase in transient failures due to AWS's eventual consistency model.
```

**Review by jillr — COMMENTED** [review]
Failing tests should just need a rebase and push; lgtm once passing

**Review by s-hertel — APPROVED** [review]

**Comment by s-hertel** [comment]
Thanks @tremble! I opened https://github.com/ansible/ansible/pull/67369 to try to address any flakiness this may have introduced. https://github.com/ansible/ansible/pull/67204 should be good to go now, yes?

**Pull request merged**
